areas/mysql/src/AzureMcp.MySql/Services/MySqlService.cs

```diff
@@ -0,0 +1,481 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text.RegularExpressions;
+using Azure;
+using Azure.Core;
+using Azure.ResourceManager.MySql.FlexibleServers;
+using AzureMcp.Core.Services.Azure;
+using AzureMcp.Core.Services.Azure.ResourceGroup;
+using AzureMcp.Core.Services.Azure.Tenant;
+using AzureMcp.MySql.Json;
+using Microsoft.Extensions.Logging;
+using MySqlConnector;
+
+namespace AzureMcp.MySql.Services;
+
+public class MySqlService(IResourceGroupService resourceGroupService, ITenantService tenantService, ILogger<MySqlService> logger) : BaseAzureService(tenantService), IMySqlService
+{
+    private readonly IResourceGroupService _resourceGroupService = resourceGroupService ?? throw new ArgumentNullException(nameof(resourceGroupService));
+    private readonly ILogger<MySqlService> _logger = logger;
+    private string? _cachedEntraIdAccessToken;
+    private DateTime _tokenExpiryTime;
+    private readonly object _tokenLock = new object();
+
+    private async Task<string> GetEntraIdAccessTokenAsync()
+    {
+        lock (_tokenLock)
+        {
+            if (_cachedEntraIdAccessToken != null && DateTime.UtcNow < _tokenExpiryTime)
+            {
+                return _cachedEntraIdAccessToken;
+            }
+        }
+
+        var tokenRequestContext = new TokenRequestContext(new[] { "https://ossrdbms-aad.database.windows.net/.default" });
+        var tokenCredential = await GetCredential();
+        var accessToken = await tokenCredential
+            .GetTokenAsync(tokenRequestContext, CancellationToken.None)
+            .ConfigureAwait(false);
+
+        lock (_tokenLock)
+        {
+            _cachedEntraIdAccessToken = accessToken.Token;
+            _tokenExpiryTime = accessToken.ExpiresOn.UtcDateTime.AddSeconds(-60); // Subtract 60 seconds as a buffer.
```

Comment:
Is this a safe assumption?

Is it possible to receive tokens with an expiry time less than 60 seconds? If so, what happens when the expiry time is negative / in the past?

Reply:
No, it is not possible to configure Azure access tokens to have an expiration time less than 60 seconds.

https://learn.microsoft.com/en-us/entra/identity-platform/configurable-token-lifetimes#access-id-and-saml2-token-lifetime-policy-properties